### ISSUE TITLE
Problem building

### ISSUE THREAD

**Issue by sonoro1234** [LuaLanes/lanes#143]
From this commit
https://github.com/LuaLanes/lanes/commit/e0dbd33c2d4776d6b2213dd82f344166eafde438

 when I try to build with cmake and mingw I get lot of warnings and error
```
[ 25%] Building C object CMakeFiles/core.dir/src/deep.c.obj
In file included from C:\luaGL\gitsources\lanes\src\deep.c:36:0:
C:\luaGL\gitsources\lanes\src\tools.h:36:34: warning: 'struct s_Universe' declared inside parameter list [enab
led by default]
 lua_State* luaG_newstate( struct s_Universe* U, lua_State* _from, char const* libs);
                                  ^
C:\luaGL\gitsources\lanes\src\tools.h:36:34: warning: its scope is only this definition or declaration, which
is probably not what you want [enabled by default]
C:\luaGL\gitsources\lanes\src\tools.h:37:42: warning: 'struct s_Universe' declared inside parameter list [enab
led by default]
 void luaG_copy_one_time_settings( struct s_Universe* U, lua_State* L, lua_State* L2);
                                          ^
C:\luaGL\gitsources\lanes\src\tools.h:48:101: warning: 'struct s_Universe' declared inside parameter list [ena
bled by default]
 char const* push_deep_proxy( struct s_Universe* U, lua_State* L, struct DEEP_PRELUDE* prelude, enum eLookupMo
de mode_);
                                                                                                     ^
C:\luaGL\gitsources\lanes\src\tools.h:51:104: warning: 'struct s_Universe' declared inside parameter list [ena
bled by default]
 int luaG_inter_copy_package( struct s_Universe* U, lua_State* L, lua_State* L2, int package_idx_, enum eLooku
pMode mode_);
                                                                                                        ^
C:\luaGL\gitsources\lanes\src\tools.h:53:88: warning: 'struct s_Universe' declared inside parameter list [enab
led by default]
 int luaG_inter_copy( struct s_Universe* U, lua_State* L, lua_State* L2, uint_t n, enum eLookupMode mode_);
                                                                                        ^
C:\luaGL\gitsources\lanes\src\tools.h:54:88: warning: 'struct s_Universe' declared inside parameter list [enab
led by default]
 int luaG_inter_move( struct s_Universe* U, lua_State* L, lua_State* L2, uint_t n, enum eLookupMode mode_);
                                                                                        ^
C:\luaGL\gitsources\lanes\src\tools.h:60:32: warning: 'struct s_Universe' declared inside parameter list [enab
led by default]
 void serialize_require( struct s_Universe* U, lua_State *L);
                                ^
C:\luaGL\gitsources\lanes\src\tools.h:61:41: warning: 'struct s_Universe' declared inside parameter list [enab
led by default]
 void initialize_on_state_create( struct s_Universe* U, lua_State* L);
                                         ^
C:\luaGL\gitsources\lanes\src\tools.h:62:87: warning: 'struct s_Universe' declared inside parameter list [enab
led by default]
 void call_on_state_create( struct s_Universe* U, lua_State* L, lua_State* from_, enum eLookupMode mode_);
                                                                                       ^
C:\luaGL\gitsources\lanes\src\deep.c:267:13: error: conflicting types for 'push_deep_proxy'
 char const* push_deep_proxy( struct s_Universe* U, lua_State* L, struct DEEP_PRELUDE* prelude, enum eLookupMo
de mode_)
             ^
In file included from C:\luaGL\gitsources\lanes\src\deep.c:36:0:
C:\luaGL\gitsources\lanes\src\tools.h:48:13: note: previous declaration of 'push_deep_proxy' was here
 char const* push_deep_proxy( struct s_Universe* U, lua_State* L, struct DEEP_PRELUDE* prelude, enum eLookupMo
de mode_);
             ^
CMakeFiles\core.dir\build.make:87: recipe for target 'CMakeFiles/core.dir/src/deep.c.obj' failed
mingw32-make[2]: *** [CMakeFiles/core.dir/src/deep.c.obj] Error 1
CMakeFiles\Makefile2:66: recipe for target 'CMakeFiles/core.dir/all' failed
mingw32-make[1]: *** [CMakeFiles/core.dir/all] Error 2
Makefile:126: recipe for target 'all' failed
mingw32-make: *** [all] Error 2
```

**Comment by sonoro1234** [comment]
Now doing PR

**Comment by benoit-germain** [comment]
Not sure about moving DEEP_PRELUDE from tools.h to universe.h.
In fact it should be in deep.h, so that's where I'm going to put it. I should be able to commit it shortly.

**Comment by benoit-germain** [comment]
Hopefull solved by 9645a86fbcb16ca83c5f4ff0dae0925ac71bb46b.

**Comment by sonoro1234** [comment]
```
[ 12%] Building C object CMakeFiles/core.dir/src/lanes.c.obj
In file included from C:\luaGL\gitsources\lanes\src\tools.h:6,
                 from C:\luaGL\gitsources\lanes\src\lanes.c:93:
C:\luaGL\gitsources\lanes\src\deep.h:45:84: warning: 'enum eLookupMode' declared inside parameter list will no
t be visible outside of this definition or declaration
 char const* push_deep_proxy( Universe* U, lua_State* L, DeepPrelude* prelude, enum eLookupMode mode_);
                                                                                    ^~~~~~~~~~~
C:\luaGL\gitsources\lanes\src\lanes.c: In function 'LG_configure':
C:\luaGL\gitsources\lanes\src\lanes.c:3164:65: error: type of formal parameter 4 is incomplete
   errmsg = push_deep_proxy( U, L, (DeepPrelude*) U->timer_deep, eLM_LaneBody);// settings M timer_deep
                                                                 ^~~~~~~~~~~~
mingw32-make[2]: *** [CMakeFiles\core.dir\build.make:63: CMakeFiles/core.dir/src/lanes.c.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:67: CMakeFiles/core.dir/all] Error 2
mingw32-make: *** [Makefile:127: all] Error 2
```
